From b96aa9b38529b6f6ba268ce72acd39cae99cb4fb Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Wed, 30 Jul 2025 10:49:23 +0200
Subject: [PATCH] ARM: dts: stm32: fix continuous display on stm32mp1 boards

Patch to fix the "continuous display" functionality.
The back-light is enabled by default, others drivers (ltdc, dsi & panels)
must detect the previous activity to maintains resources after probe.

Change-Id: I4a618a622c29ee1523030ac88357d9b42dcfe4a3
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/479222
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm/boot/dts/st/stm32mp135f-dk.dts  | 2 --
 arch/arm/boot/dts/st/stm32mp157a-ev1.dts | 4 ----
 arch/arm/boot/dts/st/stm32mp157c-dk2.dts | 2 --
 arch/arm/boot/dts/st/stm32mp157c-ev1.dts | 4 ----
 arch/arm/boot/dts/st/stm32mp157d-ev1.dts | 4 ----
 arch/arm/boot/dts/st/stm32mp157f-dk2.dts | 2 --
 arch/arm/boot/dts/st/stm32mp157f-ev1.dts | 4 ----
 7 files changed, 22 deletions(-)

diff --git a/arch/arm/boot/dts/st/stm32mp135f-dk.dts b/arch/arm/boot/dts/st/stm32mp135f-dk.dts
index d49a0ab5d8ea..5e27cd5cc94f 100644
--- a/arch/arm/boot/dts/st/stm32mp135f-dk.dts
+++ b/arch/arm/boot/dts/st/stm32mp135f-dk.dts
@@ -115,7 +115,6 @@ panel_rgb: panel-rgb {
 		backlight = <&panel_backlight>;
 		power-supply = <&scmi_v3v3_sw>;
 		data-mapping = "bgr666";
-		default-on;
 		status = "okay";
 
 		width-mm = <105>;
@@ -424,7 +423,6 @@ &ltdc {
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&ltdc_pins_a>;
 	pinctrl-1 = <&ltdc_sleep_pins_a>;
-	default-on;
 	status = "okay";
 
 	port {
diff --git a/arch/arm/boot/dts/st/stm32mp157a-ev1.dts b/arch/arm/boot/dts/st/stm32mp157a-ev1.dts
index 362db3704bb5..9bcf84f705a4 100644
--- a/arch/arm/boot/dts/st/stm32mp157a-ev1.dts
+++ b/arch/arm/boot/dts/st/stm32mp157a-ev1.dts
@@ -134,7 +134,6 @@ panel_backlight: panel-backlight {
 		compatible = "gpio-backlight";
 		gpios = <&gpiod 13 GPIO_ACTIVE_LOW>;
 		default-on;
-		default-brightness-level = <1>;
 		status = "okay";
 	};
 
@@ -340,7 +339,6 @@ dfsdm_endpoint3: endpoint {
 &dsi {
 	#address-cells = <1>;
 	#size-cells = <0>;
-	default-on;
 	status = "okay";
 
 	panel_dsi: panel@0 {
@@ -351,7 +349,6 @@ panel_dsi: panel@0 {
 		power-supply = <&v3v3>;
 		vcc-supply = <&v3v3>;
 		iovcc-supply = <&v3v3>;
-		default-on;
 		status = "okay";
 
 		port {
@@ -550,7 +547,6 @@ &i2c5 {
 };
 
 &ltdc {
-	default-on;
 	status = "okay";
 
 	port {
diff --git a/arch/arm/boot/dts/st/stm32mp157c-dk2.dts b/arch/arm/boot/dts/st/stm32mp157c-dk2.dts
index ef430bc99190..f3469e99cfc3 100644
--- a/arch/arm/boot/dts/st/stm32mp157c-dk2.dts
+++ b/arch/arm/boot/dts/st/stm32mp157c-dk2.dts
@@ -56,7 +56,6 @@ panel_otm8009a: panel-otm8009a@0 {
 		reg = <0>;
 		reset-gpios = <&gpioe 4 GPIO_ACTIVE_LOW>;
 		power-supply = <&v3v3>;
-		default-on;
 		status = "okay";
 
 		port {
@@ -91,7 +90,6 @@ touchscreen@38 {
 };
 
 &ltdc {
-	default-on;
 	status = "okay";
 
 	port {
diff --git a/arch/arm/boot/dts/st/stm32mp157c-ev1.dts b/arch/arm/boot/dts/st/stm32mp157c-ev1.dts
index 23f863ac8fa2..321346453719 100644
--- a/arch/arm/boot/dts/st/stm32mp157c-ev1.dts
+++ b/arch/arm/boot/dts/st/stm32mp157c-ev1.dts
@@ -134,7 +134,6 @@ panel_backlight: panel-backlight {
 		compatible = "gpio-backlight";
 		gpios = <&gpiod 13 GPIO_ACTIVE_LOW>;
 		default-on;
-		default-brightness-level = <1>;
 		status = "okay";
 	};
 
@@ -340,7 +339,6 @@ dfsdm_endpoint3: endpoint {
 &dsi {
 	#address-cells = <1>;
 	#size-cells = <0>;
-	default-on;
 	status = "okay";
 
 	panel_dsi: panel@0 {
@@ -351,7 +349,6 @@ panel_dsi: panel@0 {
 		power-supply = <&v3v3>;
 		vcc-supply = <&v3v3>;
 		iovcc-supply = <&v3v3>;
-		default-on;
 		status = "okay";
 
 		port {
@@ -550,7 +547,6 @@ &i2c5 {
 };
 
 &ltdc {
-	default-on;
 	status = "okay";
 
 	port {
diff --git a/arch/arm/boot/dts/st/stm32mp157d-ev1.dts b/arch/arm/boot/dts/st/stm32mp157d-ev1.dts
index 877a8c00bfbc..f584420b2007 100644
--- a/arch/arm/boot/dts/st/stm32mp157d-ev1.dts
+++ b/arch/arm/boot/dts/st/stm32mp157d-ev1.dts
@@ -134,7 +134,6 @@ panel_backlight: panel-backlight {
 		compatible = "gpio-backlight";
 		gpios = <&gpiod 13 GPIO_ACTIVE_LOW>;
 		default-on;
-		default-brightness-level = <1>;
 		status = "okay";
 	};
 
@@ -340,7 +339,6 @@ dfsdm_endpoint3: endpoint {
 &dsi {
 	#address-cells = <1>;
 	#size-cells = <0>;
-	default-on;
 	status = "okay";
 
 	panel_dsi: panel@0 {
@@ -351,7 +349,6 @@ panel_dsi: panel@0 {
 		power-supply = <&v3v3>;
 		vcc-supply = <&v3v3>;
 		iovcc-supply = <&v3v3>;
-		default-on;
 		status = "okay";
 
 		port {
@@ -550,7 +547,6 @@ &i2c5 {
 };
 
 &ltdc {
-	default-on;
 	status = "okay";
 
 	port {
diff --git a/arch/arm/boot/dts/st/stm32mp157f-dk2.dts b/arch/arm/boot/dts/st/stm32mp157f-dk2.dts
index f1cf2cbd23c3..14b3290a7b7e 100644
--- a/arch/arm/boot/dts/st/stm32mp157f-dk2.dts
+++ b/arch/arm/boot/dts/st/stm32mp157f-dk2.dts
@@ -59,7 +59,6 @@ panel_otm8009a: panel-otm8009a@0 {
 		reg = <0>;
 		reset-gpios = <&gpioe 4 GPIO_ACTIVE_LOW>;
 		power-supply = <&v3v3>;
-		default-on;
 		status = "okay";
 
 		port {
@@ -102,7 +101,6 @@ touchscreen@38 {
 };
 
 &ltdc {
-	default-on;
 	status = "okay";
 
 	port {
diff --git a/arch/arm/boot/dts/st/stm32mp157f-ev1.dts b/arch/arm/boot/dts/st/stm32mp157f-ev1.dts
index d525c2e6d352..41aa9b048f9a 100644
--- a/arch/arm/boot/dts/st/stm32mp157f-ev1.dts
+++ b/arch/arm/boot/dts/st/stm32mp157f-ev1.dts
@@ -134,7 +134,6 @@ panel_backlight: panel-backlight {
 		compatible = "gpio-backlight";
 		gpios = <&gpiod 13 GPIO_ACTIVE_LOW>;
 		default-on;
-		default-brightness-level = <1>;
 		status = "okay";
 	};
 
@@ -341,7 +340,6 @@ dfsdm_endpoint3: endpoint {
 &dsi {
 	#address-cells = <1>;
 	#size-cells = <0>;
-	default-on;
 	status = "okay";
 
 	panel_dsi: panel@0 {
@@ -352,7 +350,6 @@ panel_dsi: panel@0 {
 		power-supply = <&v3v3>;
 		vcc-supply = <&v3v3>;
 		iovcc-supply = <&v3v3>;
-		default-on;
 		status = "okay";
 
 		port {
@@ -551,7 +548,6 @@ &i2c5 {
 };
 
 &ltdc {
-	default-on;
 	status = "okay";
 
 	port {
-- 
2.53.0

